### PR TITLE
Serve/download files based on route [Not Tested]

### PR DESCRIPTION
**Pull request by kamthamc**
This PR splits files based on route. Instead of downloading everything at once, it downloads based on which page you are on.

FYI there are too many chunks

From
```
File sizes after gzip:

  579.81 KB  build/static/js/2.b5f2e2df.chunk.js
  57.22 KB   build/static/js/main.b3f69217.chunk.js
  9.62 KB    build/static/css/2.53e5fc68.chunk.css
  1.96 KB    build/static/css/main.231a5737.chunk.css
  1.39 KB    build/static/js/3.ca293cfd.chunk.js
  1.16 KB    build/static/js/runtime-main.7ba9ed1a.js
  1.02 KB    build/3a520939dabcb58f0286.worker.js
```

to 

```
File sizes after gzip:

  375.34 KB (+367.52 KB)  build/static/js/0.e0625c51.chunk.js
  127.42 KB               build/static/js/9.b0518720.chunk.js
  28.44 KB (+23.95 KB)    build/static/js/1.ccce3623.chunk.js
  16.19 KB (-63.01 KB)    build/static/js/11.9ca8263e.chunk.js
  16.08 KB (+9.31 KB)     build/static/js/29.c55b819f.chunk.js
  15.02 KB (-62.79 KB)    build/static/js/10.d8996ed9.chunk.js
  13.85 KB (-49.94 KB)    build/static/js/13.927582b1.chunk.js
  11.91 KB (-30.78 KB)    build/static/js/14.4b969c1f.chunk.js
  10.14 KB (-364.16 KB)   build/static/js/3.c5f694cd.chunk.js
  9.18 KB (+1.91 KB)      build/static/js/27.a5e55452.chunk.js
  8.85 KB (+6.7 KB)       build/static/js/main.6b2a8bf3.chunk.js
  7.28 KB (+4.19 KB)      build/static/js/2.740d950e.chunk.js
  7.08 KB (+5.19 KB)      build/static/js/32.da7d21cf.chunk.js
  6.35 KB (-8.88 KB)      build/static/js/23.8e7d7d6a.chunk.js
  6.32 KB (-87.04 KB)     build/static/js/12.dc7858c2.chunk.js
  5.45 KB (-10.57 KB)     build/static/js/25.ca181f11.chunk.js
  5.16 KB (+594 B)        build/static/js/5.6803f280.chunk.js
  4.9 KB (+2.9 KB)        build/static/css/10.19627560.chunk.css
  4.65 KB (-37.65 KB)     build/static/js/15.a516e855.chunk.js
  4.6 KB (-10.61 KB)      build/static/js/24.115de2f1.chunk.js
  4.39 KB (-46.58 KB)     build/static/js/16.c5776d27.chunk.js
  4.08 KB (-1.81 KB)      build/static/js/26.244063eb.chunk.js
  4.07 KB (-40.72 KB)     build/static/js/17.9490d9b4.chunk.js
  3.98 KB (+2.9 KB)       build/static/js/6.d9148930.chunk.js
  3.27 KB (-640 B)        build/static/js/4.3aded987.chunk.js
  3.19 KB (-13.33 KB)     build/static/js/21.32afc1ad.chunk.js
  3.17 KB (-12.01 KB)     build/static/js/22.4b0deb72.chunk.js
  3.07 KB (-2.74 KB)      build/static/js/31.7052d97c.chunk.js
  3.06 KB (-3.95 KB)      build/static/js/30.cb726fab.chunk.js
  3.06 KB (-3.67 KB)      build/static/js/28.3748156e.chunk.js
  3.05 KB (+1.15 KB)      build/static/js/33.69c119fb.chunk.js
  2.83 KB                 build/static/css/13.52ad7489.chunk.css
  2.82 KB (-35.93 KB)     build/static/js/19.78943a03.chunk.js
  2.79 KB (-14.44 KB)     build/static/js/20.747fa7e5.chunk.js
  2.79 KB (-44.47 KB)     build/static/js/18.9ff637ef.chunk.js
  2 KB                    build/static/css/9.e94f92bb.chunk.css
  1.96 KB                 build/static/css/main.231a5737.chunk.css
  1.86 KB (-67 B)         build/static/js/runtime-main.10ecffc2.js
  1.41 KB (-3.27 KB)      build/static/js/35.69e003bf.chunk.js
  1.4 KB (-3.27 KB)       build/static/js/36.5e37930a.chunk.js
  1.38 KB (-538 B)        build/static/js/34.be8e59eb.chunk.js
  1.02 KB                 build/3a520939dabcb58f0286.worker.js
```

### CODE REVIEW

**Comment by NdibeRaymond** [comment]
Hello @kamthamc , thanks for this PR! I will take time to test it then merge it. Thanks again!

File: zubhub_frontend/zubhub/src/App.js [hunk 1]
```diff
@@ -1,34 +1,75 @@
...
+const StaffPickDetails = React.lazy(() =>
+  import('./views/staff_pick_details/StaffPickDetails'),
+);
+const Guidelines = React.lazy(() => import('./views/guidelines/Guidelines'));
+const TermsOfUse = React.lazy(() => import('./views/terms_of_use/TermsOfUse'));
+const About = React.lazy(() => import('./views/about/About'));
+const FAQs = React.lazy(() => import('./views/faqs/FAQs'));
+
+const LazyImport = props => {
+  const { LazyComponent, restOfProps } = props;
```
Comment (NdibeRaymond) [inline zubhub_frontend/zubhub/src/App.js]: Hello @kamthamc cc @tuxology , I got around to testing this and it seems that we have one very little bug. the line `const {lazyComponent, restOfProps } = props` is not using rest parameter for "restOfProps" and therefore misses to pass some props along. When we try to use those props ("t" for example for i18n), the code breaks. that line should be `const {lazyComponent, ...restOfProps } = props` instead.

I can checkout this commit locally, fix it and recreate a new PR if you don't have the time to review this, but I figured I should mention it first and see what you think

**Review by NdibeRaymond — CHANGES_REQUESTED** [review]

File: zubhub_frontend/zubhub/src/App.js (hunk 1 above)
Comment (kamthamc) [reply zubhub_frontend/zubhub/src/App.js]: sorry I do not understand, `restOfProps` should have both `routeProps` and App `props` and its passed to the LazyComponent in Ln69
`<LazyComponent {...restOfProps} />`

Comment (NdibeRaymond) [reply zubhub_frontend/zubhub/src/App.js]: Ok let me explain with images:

# with this 
cause_of_t_is_not_a_function

# we get this:
t_is_not_a_function
(t here is an i18n function for fetching language strings of a particular key)

# With this:
cause_of_normal_operation


# We get this:
normal operation

Comment (kamthamc) [reply zubhub_frontend/zubhub/src/App.js]: Oh sorry my bad it was supposed to be …restOfProps

**Comment by kamthamc** [comment]
The props are fixed now

**Comment by NdibeRaymond** [comment]
> The props are fixed now

Ok thanks @kamthamc , merging now

**Review by NdibeRaymond — APPROVED** [review]

**Pull request merged**